### DISCUSSION TITLE
Cassandra Installation

### DISCUSSION THREAD

**Discussion by arrennbaral** [Homebrew/discussions#687] [Everyday usage]
Unsuccessful Cassandra Installation in new M1 Mac.

I tried to install cassandra as Terminal's suggested **brew install --build-from-source cassandra**
It is successfully installed, but I cannot start/stop cassandra after the installation.(check at the bottom)
I dont know if i am missing something or others are going through the same.
For some reason, 
brew install Cassandra doesn't work on my machine.

@macbook-pro ~ % brew install --build-from-source cassandra
==> Downloading https://raw.githubusercontent.com/Homebrew/formula-patches/9976a
Already downloaded: /Users/arrennbaral/Library/Caches/Homebrew/downloads/d8844da849e8898f5db143aabb6f28de2d5fe7e8d124afaa3c543a5d54b3163d--xcode.patch
==> Downloading https://cdn.azul.com/zulu/bin/zulu7.42.0.13-ca-jdk7.0.282-macosx
Already downloaded: /Users/arrennbaral/Library/Caches/Homebrew/downloads/4b249391d5268af07ee53bf92b453584ae6ed2ba6680bc5a22a6179a6cdd3f74--zulu7.42.0.13-ca-jdk7.0.282-macosx_x64.tar.gz
==> Downloading https://openjdk-sources.osci.io/openjdk8/openjdk8u282-ga.tar.xz
Already downloaded: /Users/arrennbaral/Library/Caches/Homebrew/downloads/ada8bfa0597910ef687e1b313915c0b3d81d0a75babe0de6d9e0a889b7d4caf8--openjdk8u282-ga.tar.xz
==> Downloading https://files.pythonhosted.org/packages/b2/40/4e00501c204b457f10
Already downloaded: /Users/Library/Caches/Homebrew/downloads/96184832975d570c3f54c37496b73256589ffb7c33c65edc3542bc2b474b6951--setuptools-44.1.1.zip
==> Downloading https://files.pythonhosted.org/packages/47/04/5fc6c74ad114032cd2
Already downloaded: /Users//Library/Caches/Homebrew/downloads/991138ddcf40267021cd87f35a0573287201a890bd378ccdbb122b50334b258d--futures-3.3.0.tar.gz
==> Downloading https://files.pythonhosted.org/packages/6b/34/415834bfdafca3c5f4
Already downloaded: /Users/l/Library/Caches/Homebrew/downloads/e99698a932bde9f5fb8ca552923ae339984f284e1c7764df95c12175f573c0d7--six-1.15.0.tar.gz
==> Downloading https://files.pythonhosted.org/packages/ae/58/35e3f0cd290039ff86
Already downloaded: /Users//Library/Caches/Homebrew/downloads/e27d635b5d2d6c3612e7f8b8a20f7fcbe75427972caaf8809ef2ac7bbf905be2--thrift-0.9.3.tar.gz
==> Downloading https://files.pythonhosted.org/packages/0b/15/523f6008d32f05dd3c
Already downloaded: /Users//Library/Caches/Homebrew/downloads/e85631dbbf140427c0cd27288ef13e53dc194138dd4dbb7161a61fb175c1817e--cql-1.4.0.tar.gz
==> Downloading https://files.pythonhosted.org/packages/cd/22/7bf65cfd5d60f3c916
Already downloaded: /Users//Library/Caches/Homebrew/downloads/f6e2fcb9e1a53ce4304530294a61f7923d764f17e78a02d873f4203342cd2c88--cassandra-driver-3.24.0.tar.gz
==> Downloading https://www.apache.org/dyn/closer.lua?path=cassandra/3.11.10/apa
==> Downloading from https://mirrors.gigenet.com/apache/cassandra/3.11.10/apache
######################################################################## 100.0%
==> Installing dependencies for cassandra: openjdk@8
==> Installing cassandra dependency: openjdk@8
==> Patching
==> Applying xcode.patch
patching file common/autoconf/basics.m4
patching file common/autoconf/flags.m4
patching file common/autoconf/generated-configure.sh
Hunk #1 succeeded at 1897 (offset 2 lines).
Hunk #2 succeeded at 14009 (offset 1 line).
Hunk #3 succeeded at 26774 (offset 23 lines).
Hunk #4 succeeded at 26837 (offset 23 lines).
Hunk #5 succeeded at 41295 (offset 23 lines).
Hunk #6 succeeded at 41394 (offset 23 lines).
Hunk #7 succeeded at 41420 (offset 23 lines).
Hunk #8 succeeded at 41485 (offset 23 lines).
Hunk #9 succeeded at 42170 (offset 23 lines).
Hunk #10 succeeded at 42259 (offset 23 lines).
Hunk #11 succeeded at 48934 (offset 23 lines).
patching file common/autoconf/libraries.m4
patching file common/autoconf/platform.m4
patching file common/autoconf/toolchain.m4
patching file hotspot/make/bsd/makefiles/gcc.make
patching file hotspot/make/bsd/makefiles/jsig.make
patching file hotspot/src/os/aix/vm/decoder_aix.hpp
patching file hotspot/src/os/bsd/vm/decoder_machO.hpp
patching file hotspot/src/os/windows/vm/decoder_windows.hpp
patching file hotspot/src/share/vm/classfile/classLoader.hpp
patching file hotspot/src/share/vm/prims/jvmtiImpl.hpp
patching file hotspot/src/share/vm/runtime/perfData.hpp
patching file hotspot/src/share/vm/utilities/decoder.hpp
patching file hotspot/src/share/vm/utilities/decoder_elf.hpp
patching file jdk/make/lib/CoreLibraries.gmk
Hunk #1 succeeded at 409 (offset -1 lines).
Hunk #2 succeeded at 443 (offset -1 lines).
Hunk #3 succeeded at 504 (offset -1 lines).
patching file jdk/src/macosx/native/sun/osxapp/ThreadUtilities.m
==> common/autoconf/autogen.sh
==> ./configure --with-boot-jdk-jvmargs=-Duser.home=/Users/arrennbaral/Library/C
Last 15 lines from /Users/arrennbaral/Library/Logs/Homebrew/openjdk@8/02.configure:
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking stdio.h usability... yes
checking stdio.h presence... yes
checking for stdio.h... yes
checking size of int *... 8
configure: The tested number of bits in the target (64) differs from the number of bits expected to be found in the target (32).
configure: I'll retry after setting the platforms compiler target bits flag to -m32
checking size of int *... 0
configure: error: The tested number of bits in the target (0) differs from the number of bits expected to be found in the target (32)
configure exiting with result code 1

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
openjdk 15.0.2 #69460
openjdk@11 11.0.10 #69396
OpenJDK post-install fails without Xcode on M1 #69100
Cassandra 3.11.9_1 crashes in openjdk@8 JVM #66462
OpenJDK is somewhat broken on newer MacOS instances, console is flooded with errors when using JMeter, AdoptOpenJDK has no issues #66953

@macbook-pro ~ % brew services start cassandra
Error: Formula cassandra is not installed.

**Comment by SMillerDev** [comment]
That's expected. Not all software works on ARM. 

---
Please don't post your issue multiple times, your issue is really not important enough to waste maintainers time looking at it multiple times.

**Reply by arrennbaral** [reply]
Thank you for the response, SMillerDev.

**Comment by abrarshahid** [comment]
+1 for M1 Mac support
